pkg/v1/openapischema/schema_default.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package openapischema provides utilities for working with openapi schema
+package openapischema
+
+import (
+	"bytes"
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// SchemaDefault returns a yaml byte array with values populated according to schema
+func SchemaDefault(schema []byte) ([]byte, error) {
+	jsonSchemaProps := &apiextensions.JSONSchemaProps{}
+
+	if err := yaml.Unmarshal(schema, jsonSchemaProps); err != nil {
+		return nil, err
+	}
+	s, err := structuralschema.NewStructural(jsonSchemaProps)
+	if err != nil {
+		return nil, err
+	}
+	unstructured := make(map[string]interface{})
+
+	schemaDefault(unstructured, s)
+	var b bytes.Buffer
+	yamlEncoder := yaml.NewEncoder(&b)
+	yamlEncoder.SetIndent(2)
+	if err := yamlEncoder.Encode(unstructured); err != nil {
+		return nil, err
+	}
+	if err := yamlEncoder.Close(); err != nil {
+		return nil, err
+	}
+
+	return b.Bytes(), nil
+}
+
+// schemaDefault does defaulting of x depending on default values in s.
+// This is adopted from k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting with 2 changes
+// 1. Prevent deep copy of int as it panics
```

Comment:
Is this a bug in upstream https://github.com/kubernetes/apiextensions-apiserver/blob/master/pkg/apiserver/schema/defaulting/algorithm.go ?

Reply:
Possibly but not sure.